plugins/org.jkiss.dbeaver.erd.ui/src/org/jkiss/dbeaver/erd/ui/action/DiagramExportAction.java

```diff
@@ -24,22 +24,59 @@
 import org.jkiss.dbeaver.erd.ui.editor.ERDEditorEmbedded;
 import org.jkiss.dbeaver.erd.ui.internal.ERDUIActivator;
 
+import org.eclipse.ui.handlers.HandlerUtil;
+import org.eclipse.swt.widgets.Composite;
+import org.eclipse.swt.widgets.FileDialog;
+import org.eclipse.swt.SWT;
+import org.jkiss.dbeaver.Log;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.OutputStreamWriter;
+import java.util.HashMap;
+import org.jkiss.dbeaver.utils.GeneralUtils;
+import org.jkiss.dbeaver.utils.RuntimeUtils;
+import org.jkiss.dbeaver.erd.ui.model.DiagramLoader;
+import org.jkiss.dbeaver.erd.ui.editor.ERDEditorPart;
+
 /**
  * Action to toggle diagram persistence
  */
 public class DiagramExportAction extends Action {
+
+    private static final Log log = Log.getLog(DiagramExportAction.class);
+
     private final ERDEditorEmbedded editor;
+    private Composite parent;
 
-    public DiagramExportAction(ERDEditorEmbedded editor) {
+    public DiagramExportAction(ERDEditorEmbedded editor, Composite parent) {
         super("Save diagram", ERDUIActivator.getImageDescriptor("icons/arrangeall.png"));
         setDescription("Save diagram into a ERD file");
         setToolTipText(getDescription());
         this.editor = editor;
+        this.parent = parent;
     }
 
     @Override
     public void run() {
-        editor.saveErdToFile(new NullProgressMonitor());
+        try {
+            String path = null;
+            FileDialog dialog = new FileDialog( parent.getShell(), SWT.SAVE );
+            String[] filterExt = {"*.erd"};
+            dialog.setFilterExtensions(filterExt);
+            path = dialog.open();
+            
+            if ( path != null ) {
+                String diagramState = DiagramLoader.serializeDiagram(RuntimeUtils.makeMonitor(new NullProgressMonitor()), editor.getDiagramPart(), editor.getDiagram(), false, true);
```

Comment:
As I already said, use `new VoidProgressMonitor()` instead of `RuntimeUtils.makeMonitor(new NullProgressMonitor())`